File: src/benchmarks/go/closurealloc.go
package main

func test() func() *[]int {

	shared := []int{}
	shared = append(shared, 0)

	var ret = func() *[]int {
		return &shared
	}

	return ret
}

func main() {
	for i := 0; i < 1000000; i++ {
		test()()
	}
}

File: src/benchmarks/c/closurealloc.c
#include <stdlib.h>
#include <stdio.h>

typedef int (*f)();

void *shared;

int ret()
{
    return (int)shared;
}

f test()
{
    shared = malloc(5);
    return ret;
}

int main()
{

    for (int i = 0; i < 1000000; i++)
        test()();
    free(shared);
}